### DISCUSSION TITLE
TransactionTracer

### DISCUSSION THREAD

**Discussion by sgtsquiggs** [jackc/pgx#1750] [Ideas]
I am trying to add more detailed tracing w/ otel and datadog and currently I cannot easily trace transactions without modifying pgx. Was there any prior work for a transaction tracer I can look at?

**Comment by jackc** [comment]
Not that I know of.

However, transactions are executed by calling `Exec` so `QueryTracer` will at least get triggered on the `begin`, `commit`, and `rollback` calls.
